tests/Unit/message.test.js

```diff
@@ -0,0 +1,290 @@
+      expect(error).toEqual(err);
+    };
+    await functions.deleteMessage(req, res, next);
+    expect(hasError).toBe(true);
+  });
+});
+
+describe("edit message tests", () => {
+  test("normal edit message", async () => {
+    const body = {
+      ID: "5dd4dcfe36424d441068f7aa",
+      text: "sdfw",
+    };
+    const req = {
+      body,
+    };
+
+    const res = {
+      json: (ret) => {
+        expect(ret.query._id).toEqual(body.ID);
+        expect(ret.update.text).toEqual(body.text);
+      },
+    };
+    const next = {};
+    await functions.editMessage(req, res, next);
+  });
+
+  test("edit message not found", async () => {
+    var hasError = false;
+    const body = {
+      ID: "5dd4dcfe36424d441068faa",
+    };
+    const req = {
+      body,
+    };
+    const res = {
+      json: (ret) => {},
+    };
+    const next = (error) => {
+      hasError = true;
+      const err = new Error(errors.MESSAGE_NOT_EXIST);
+      err.name = errors.MESSAGE_NOT_EXIST;
+      expect(error).toEqual(err);
+    };
+    await functions.editMessage(req, res, next);
+    expect(hasError).toBe(true);
+  });
+
+  test("edit message no text", async () => {
+    var hasError = false;
+    const body = {
+      ID: "5dd4dcfe36424d441068f7aa",
+      text: "",
+    };
+    const req = {
+      body,
+    };
+
+    const res = {
+      json: (ret) => {},
+    };
+    const next = (error) => {
+      hasError = true;
+      const err = new Error(errors.MESSAGE_NO_TEXT);
+      err.name = errors.MESSAGE_NO_TEXT;
+      expect(error).toEqual(err);
+    };
+    await functions.editMessage(req, res, next);
+    expect(hasError).toBe(true);
+  });
+
+  test("edit message not allowed", async () => {
+    var hasError = false;
+    const body = {
+      ID: "5dd4dcfe36424d441068faa",
+    };
+    const req = {
+      body,
+    };
+    const res = {
+      json: (ret) => {},
+    };
+    const next = (error) => {
+      hasError = true;
+      const err = new Error(errors.PLEASE_AUTHENTICATE);
+      err.name = errors.PLEASE_AUTHENTICATE;
+      expect(error).toEqual(err);
+    };
+    await functions.editMessage(req, res, next);
+    expect(hasError).toBe(true);
+  });
+});
+
+describe("get message tests", () => {
+  test("normal get message", async () => {
+    const body = {
+      ID: "5dd4dcfe36424d441068f7aa",
+    };
+    const req = {
+      body,
+    };
+    const res = {
+      json: (ret) => {
+        expect(ret._id).toEqual(body.ID);
+      },
+    };
+    const next = (error) => {};
+    await functions.getMessage(req, res, next);
+  });
+
+  test("get message not found", async () => {
+    var hasError = false;
+    const body = {
+      ID: "5dd4dcfe36424d441068faa",
+    };
+    const req = {
+      body,
+    };
+    const res = {
+      json: (ret) => {},
+    };
+    const next = (error) => {
+      hasError = true;
+      const err = new Error(errors.MESSAGE_NOT_EXIST);
+      err.name = errors.MESSAGE_NOT_EXIST;
```

Comment:
This should use 404 instead of bad request

Reply:
MESSGE_NOT_EXIST is not found tho